### DISCUSSION TITLE
Why does the NetData Docker need so much access to the host?

### DISCUSSION THREAD

**Discussion by SuperPauly** [netdata/netdata#16313] [Q&A] [answered]
Looking at the instructions at https://learn.netdata.cloud/docs/installing/docker unless I'm missing something? To me this seems a bit intrusive for a metrics collection webserver. 

The whole point of Docker is a container away from the host system. But here you are using --pid host which means this docker can see all running processes of the host, it is also mount the /run dir of the host again being able to see all running processes.

Why do you need to use --cap SYS_ADMIN? Being able to run PTrace means you can see any process running live on my host with any encrypted data like passwords or locations of private certificates/keys. 

I don't really like the idea you need to be out of the container and on my host network with using  --network=host. Again defeating the point of a container. Why would I want you software on my host's network? If I do want it to have access to my hosts network I will add the rules rather than just handing over the keys.

This could widen the scope for a bunch of severe vuln with these types of permissions.

I don't understand why the need for so much control/access over the host?

It almost seems pointless as having this as a containerised system.

It seems like a awesome piece of software but this is stopping me from installing it.

Do I really need them all?

**Comment by ilyam8** [comment]
It is literally described in the first section of the documentation you are referring to.

Let me highlight the gist

> The Netdata container requires different privileges and mounts **to provide functionality similar to that provided by Netdata installed on the host**.

Hence the [Recommended Way](https://learn.netdata.cloud/docs/installing/docker#recommended-way) has all privileges and mounts to have more or less feature parity with Netdata installed on the host.

You are free to remove any of them, but you will lose some functionality/metrics (described in that doc).

By default (Recommended Way), we treat docker as a packaging method, not a secure installation.

**Reply by ilyam8** [reply]
What we can do - add a sample config with minimal privileges and mounts to the list

<img width="315" alt="Screenshot 2023-10-31 at 18 08 14" src="https://github.com/netdata/netdata/assets/22274335/e2f4cf70-acd5-4655-baaf-cac4e4bb3573">

But it won't be recommended.

**Reply by SuperPauly** [reply]
I see, this might be best installed on a VM for my use case then. If a container did have minimal privileges what effect would it have on the service?

**Reply by ilyam8** [reply, marked as answer]
Expand Privileges and Mounts, we have a user/description for every item.

<img width="960" alt="Screenshot 2023-10-31 at 18 52 03" src="https://github.com/netdata/netdata/assets/22274335/fbb1d7d8-188e-4e73-b588-26d56eb3435b">
